x-pack/plugins/siem/public/store/timeline/epic.ts

```diff
@@ -0,0 +1,302 @@
+        } else if (timelineFavoriteActionsType.includes(action.type)) {
+          return epicPersistTimelineFavorite(apolloClient, action, timeline, action$, timeline$);
+        } else if (timelineActionsType.includes(action.type)) {
+          return from(
+            apolloClient.mutate<
+              PersistTimelineMutation.Mutation,
+              PersistTimelineMutation.Variables
+            >({
+              mutation: persistTimelineMutation,
+              fetchPolicy: 'no-cache',
+              variables: {
+                timelineId,
+                version,
+                timeline: convertTimelineAsInput(
+                  timeline[get('payload.id', action)],
+                  timelineTimeRange
+                ),
+              },
+              refetchQueries,
+            })
+          ).pipe(
+            withLatestFrom(timeline$),
+            mergeMap(([result, recentTimeline]) => {
+              const savedTimeline = recentTimeline[get('payload.id', action)];
+              const response: ResponseTimeline = get('data.persistTimeline', result);
+
+              return [
+                response.code === 409
+                  ? updateAutoSaveMsg({
+                      timelineId: get('payload.id', action),
+                      newTimelineModel: omitTypenameInTimeline(savedTimeline, response.timeline),
+                    })
+                  : updateTimeline({
+                      id: get('payload.id', action),
+                      timeline: {
+                        ...savedTimeline,
+                        savedObjectId: response.timeline.savedObjectId,
+                        version: response.timeline.version,
+                        isSaving: false,
+                      },
+                    }),
+                endTimelineSaving({
+                  id: get('payload.id', action),
+                }),
+              ];
+            }),
+            startWith(startTimelineSaving({ id: get('payload.id', action) })),
+            takeUntil(
+              action$.pipe(
+                withLatestFrom(timeline$),
+                filter(([checkAction, updatedTimeline]) => {
+                  if (
+                    checkAction.type === endTimelineSaving.type &&
+                    updatedTimeline[get('payload.id', checkAction)].savedObjectId != null
+                  ) {
+                    myEpicTimelineId.setTimelineId(
+                      updatedTimeline[get('payload.id', checkAction)].savedObjectId
+                    );
+                    myEpicTimelineId.setTimelineVersion(
+                      updatedTimeline[get('payload.id', checkAction)].version
+                    );
+                    return true;
+                  }
+                  return false;
+                })
+              )
+            )
+          );
+        }
+        return empty();
+      })
+    )
+  );
+};
+
+const isNotNull = <T>(value: T | null): value is T => value !== null;
```

Comment:
You already have this `isNotNull` in: `x-pack/plugins/siem/public/store/timeline/helpers.ts`, I would just use that one if you could to avoid duping it again.